File: sorting/selectionSort/selectionSort.java
import java.util.Arrays;

public class selectionSort {

    static void selectionSortFn(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int minimun = i;
            for (int j = i+1; j < arr.length; j++) {
                if (arr[j] < arr[minimun]) {
                    minimun = j;
                }
            }

            int temp = arr[i];
            arr[i] = arr[minimun];
            arr[minimun] = temp;
            System.out.println(Arrays.toString(arr)  + " one iteration");
        }
    }

    public static void main(String[] args) {
        selectionSortFn(new int[]{2,5,7,1,3,34,55,22,54,21,5,7});
    }
}

File: namasteDSA/sorting/selectionSort.js
function selectionSort(arr){
    let n = arr.length;
    for(let i=0;i<n-1;i++){
        let min = i;
        for(let j =i+1;j<n;j++){
            if(arr[min]>arr[j]){
                min=j;
            }
        }
        if(min!=i){
            swap(arr,i,min)
        }
    }
    return arr;
}

function swap(arr,i,j){
    return [arr[i],arr[j]]=[arr[j],arr[i]]
}

console.log(selectionSort([5,2,4,6,7,82,3,5,7,1,1,3,5,]))